narwhal/primary/src/proposer.rs

```diff
@@ -203,6 +216,7 @@ impl Proposer {
         // Make a new header.
         let num_of_digests = self.digests.len().min(self.max_header_num_of_batches);
         let header_digests: VecDeque<_> = self.digests.drain(..num_of_digests).collect();
+        let system_messages: Vec<_> = self.system_messages.drain(..).collect();
```

Comment:
hopefully compiles to the same thing, but it hurts my feelings that we aren't just using `std::mem::take` here

Reply:
on the off chance that the existing code is actually doing multiple allocations, we might as well. `take` is also more readable imo